algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -173,11 +176,11 @@ impl<E: PairingEngine, S: AlgebraicSponge<E::Fq, 2>> SonicKZG10<E, S> {
     /// polynomial will have the corresponding degree bound enforced.
     #[allow(clippy::type_complexity)]
     #[allow(clippy::format_push_string)]
-    pub fn commit<'b>(
+    pub fn commit<'b, R: Rng + CryptoRng>(
         universal_prover: &UniversalProver<E>,
         ck: &CommitterUnionKey<E>,
         polynomials: impl IntoIterator<Item = LabeledPolynomialWithBasis<'b, E::Fr>>,
-        rng: Option<&mut dyn RngCore>,
+        rng: Option<&mut R>,
```

Comment:
Why this change?

Reply:
Hmm it did not compile during an intermediary test, now it does: e791de587

Reply:
Just for some context to understand the situation:

It's better not to use dynamic dispatch when possible. The perf hit can be significant depending on context and frequency of use. I think this case would be innocent, but nonetheless not ideal.

Reply:
I think it's fine, because we only sample a couple of field elements from the rng; it's better than the alternative because when rng is `None`, it doesn't require us to specify the type of the (unused) `Rng`